File: src/main/java/edu/dcccd/trans/controller/RESTController.java
package edu.dcccd.trans.controller;

import edu.dcccd.trans.service.JokeService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class RESTController {
    private final JokeService jokeService;

    @Autowired
    public RESTController( JokeService jokeService ) {
        this.jokeService = jokeService;
    }

    @GetMapping(value = ("/joke"))
    public String getJoke() throws Exception {
        return jokeService.getJoke("Craig", "Walls");
    }
}

File: src/main/kotlin/edu/dcccd/trans/controller/RESTController.kt
package edu.dcccd.trans.controller

import edu.dcccd.trans.service.JokeService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class RESTController @Autowired
constructor(private val jokeService: JokeService) {
    val joke: String?
        @GetMapping(value = ["/joke"])
        @Throws(Exception::class)
        get() = jokeService.getJoke("Craig", "Walls")
}
